[jit] Simplify recursive script compilation flow.

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31020 [jit] avoid doing quadratic work in concrete type inference
* **#31019 [jit] Simplify recursive script compilation flow.**
* #31018 [jit] remove remnants of properties hack
* #31017 [jit] remove unnecessary arg from create_script_module

No more `recurisve_script`, just direct calls to `create_script_module`.
This reduces the number of pathways through the frontend, and the
uniformity is useful for a future PR.

Differential Revision: [D18904113](https://our.internmc.facebook.com/intern/diff/D18904113)

suo committed:
[jit] Simplify recursive script compilation flow.

No more `recurisve_script`, just direct calls to `create_script_module`.
This reduces the number of pathways through the frontend, and the
uniformity is useful for a future PR.

suo committed:
Update on "[jit] Simplify recursive script compilation flow."


No more `recurisve_script`, just direct calls to `create_script_module`.
This reduces the number of pathways through the frontend, and the
uniformity is useful for a future PR.

kostmo commented:
## CircleCI build failures summary
As of commit bb6dca8:
* **1/1** failures introduced in this PR



## Detailed failure analysis (WIP)

Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=bb6dca884d158088d827761cf469e01a3e4bf971):

### 1 failure recognized by patterns:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3873247) pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build (1/1)
**Step:** pytorch android gradle build ([details](https://dr.pytorch.org/build-details.html?build_id=71135084))
```
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-bb6dca884d158088d827761cf469e01a3e4bf971-android-arm-v7a not found
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

zdevito commented on torch/jit/_recursive.py:
```diff
@@ -258,51 +258,43 @@ def create_methods_from_stubs(concrete_type, stubs):
     Arguments:
         nn_module:  The original Python nn.Module that we are creating a ScriptModule for.
-        stubs:  ScriptMethodStubs to compile as part of the conversion process.
+        stubs_fn:  Lambda that takes an nn.Module and generates a list of ScriptMethodStubs to compile.
```
nit: this lambda looks like it is called unconditionally, so why not just pass the list of stubs?

suo replied:
If `nn_module` is a scriptmodule, it won't be called and we'll just use the scriptmodule as-is.

zdevito reviewed (approved):
Nice cleanup!

suo committed:
Update on "[jit] Simplify recursive script compilation flow."


No more `recurisve_script`, just direct calls to `create_script_module`.
This reduces the number of pathways through the frontend, and the
uniformity is useful for a future PR.

Differential Revision: [D18904113](https://our.internmc.facebook.com/intern/diff/D18904113)
